lib/reline/line_editor.rb

```diff
@@ -2664,7 +2666,7 @@ def finish
   alias_method :kill_whole_line, :em_kill_line
 
   private def em_delete(key)
-    if (not @is_multiline and @line.empty?) or (@is_multiline and @line.empty? and @buffer_of_lines.size == 1)
+    if @line.empty? and (not @is_multiline or @buffer_of_lines.size == 1) and key == 4
```

Comment:
We could change 4 to `"\C-D".ord` for more clarity.

Reply:
Good idea, thanks! Also rebased the branch.